examples/big-numbers/README.md

```diff
@@ -0,0 +1,260 @@
+    let eth1 = U256::from(10_000000000000000000_u128); // 10 ether
+    let eth2 = U256::from(20_000000000000000000_u128); // 20 ether
+    let base = U256::from(10).pow(18.into());
+    let mul = eth1.mul(eth2).div(base); // We also divide by 10^18
+    let s: String = format_units(mul, "ether").unwrap();
+    assert_eq!(&*s, "200.000000000000000000"); // 200
+}
+```
+
+## Utilities
+In order to create an application, it is often necessary to convert between the representation of values that is easily understood by humans (such as ether) and the machine-readable form that is used by contracts and math functions (such as wei). This is particularly important when working with Ethereum, as certain values, such as balances and gas prices, must be expressed in wei when sending transactions, even if they are displayed to the user in a different format, such as ether or gwei. To help with this conversion, ethers-rs provides two functions, `parse_units` and `format_units`, which allow you to easily convert between human-readable and machine-readable forms of values. `parse_units` can be used to convert a string representing a value in ether, such as "1.1", into a big number in wei, which can be used in contracts and math functions. `format_units` can be used to convert a big number value into a human-readable string, which is useful for displaying values to users.
+
+### `parse_units`
+```rust
+use ethers::{types::U256, utils::{parse_units, ParseUnits}};
+
+fn main() {
+    let pu: ParseUnits = parse_units("1.0", "wei").unwrap();
+    let num = U256::from(pu);
+    assert!(num.eq(&U256::from(1)));
+
+    let pu: ParseUnits = parse_units("1.0", "kwei").unwrap();
+    let num = U256::from(pu);
+    assert!(num.eq(&U256::from(1000)));
+
+    let pu: ParseUnits = parse_units("1.0", "mwei").unwrap();
+    let num = U256::from(pu);
+    assert!(num.eq(&U256::from(1000000)));
+
+    let pu: ParseUnits = parse_units("1.0", "gwei").unwrap();
+    let num = U256::from(pu);
+    assert!(num.eq(&U256::from(1000000000)));
+
+    let pu: ParseUnits = parse_units("1.0", "szabo").unwrap();
+    let num = U256::from(pu);
+    assert!(num.eq(&U256::from(1000000000000_u128)));
+
+    let pu: ParseUnits = parse_units("1.0", "finney").unwrap();
+    let num = U256::from(pu);
+    assert!(num.eq(&U256::from(1000000000000000_u128)));
+
+    let pu: ParseUnits = parse_units("1.0", "ether").unwrap();
+    let num = U256::from(pu);
+    assert!(num.eq(&U256::from(1000000000000000000_u128)));
+
+    let pu: ParseUnits = parse_units("1.0", 18).unwrap();
+    let num = U256::from(pu);
+    assert!(num.eq(&U256::from(1000000000000000000_u128)));
+
+}
+```
+
+### `format_units`
+```rust
+use ethers::{types::U256, utils::format_units};
+
+fn format_units_example() {
+    // 1 ETHER = 10^18 WEI
+    let one_ether = U256::from(1000000000000000000_u128);
+
+    let num: String = format_units(one_ether, "wei").unwrap();
+    assert_eq!(&*num, "1000000000000000000.0");
```

Comment:
I don't think the `&*` are necessary when doing `assert_eq!` for `String` with `str`, but I may be wrong; in that case the idiomatic way is to use `as_str`.

Reply:
Thank you for the hint. Now it's much cleaner.